app/code/core/Mage/Adminhtml/Block/Catalog/Category/Abstract.php

```diff
@@ -134,6 +134,7 @@ public function getRoot($parentNodeCategory=null, $recursionLevel=3)
      * If ids are empty, default tree with depth = 2 will be returned.
      *
      * @param array $ids
+     * @return mixed|Varien_Data_Tree_Node|null
```

Comment:
Checking all the [`Mage::register('root', $root);`](https://github.com/OpenMage/magento-lts/search?q=Mage::register(%27root%27,%20$root);), I only see that `$root` can only be

```suggestion
     * @return Varien_Data_Tree_Node|null
```

No?

Reply:
No! :)

PhpStan just knows about this ...

```
    /**
     * Retrieve a value from registry by a key
     *
     * @param string $key
     * @return mixed
     */
    public static function registry($key)
```

It does not know the content of `$root`.

Reply:
Tricky, I see it now. Thanks

Reply:
@kiatng another examples is`@param Varien_Object|Same_Name` ... in most cases `Varien_Object` is to match a parent methods return type or typehints. `Same_Name` is used to look up for child classes and their methods.